frontend/src/metabase/dashboard/containers/ArchiveDashboardModal.jsx

```diff
@@ -23,25 +23,41 @@ class ArchiveDashboardModal extends Component {
     onClose: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = { loading: false };
+  }
+
   archive = async () => {
     const dashboardId = Urls.extractEntityId(this.props.params.slug);
+    this.setState({ loading: true });
     await this.props.setDashboardArchived(dashboardId);
+    this.setState({ loading: false });
```

Comment:
I think we need to consider a failure state here.

Reply:
The failure state is handled in `ArchiveModal.tsx`, though we may never set loading back to false if there is an error :/